framework/Distributions.py

```diff
@@ -1752,7 +1744,8 @@ def ppf(self,x):
     cumulative=0.0
     for element in sortedMapping:
       cumulative += element[1]
-      if cumulative >= x:
+      #if comulative >= x - np.finfo(float).eps:
+      if cumulative >= (x*self.totPsum):
```

Comment:
This is where @wangcj05 can provide some suggestion

Reply:
I suggest to perform a normalization in the initializeDistribution method. Basically, the value stored in self.mapping will be normalized.